src/app/components/header/header.component.ts

```diff
@@ -17,7 +19,9 @@ import { showLoginModal } from '../../store/modal.actions';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class HeaderComponent {
+  private router = inject(Router)
```

Comment:
please change to constructor DI it has more opportunity
```suggestion
  private router: Router
```